ProductApi/Model/Api/SkuAvailability.php

```diff
@@ -22,6 +54,52 @@ public function getAvailability(
         $sku,
         $storeSourceCode
     ) {
+        $skuFilter = $this->filterBuilder
+            ->setField('sku')
+            ->setConditionType('eq')
+            ->setValue($sku)
+            ->create();
+
+        $filterGroup1 = $this->filterGroupBuilder
+            ->setFilters([$skuFilter])
+            ->create();
+
+        $sourceCodeFilter = $this->filterBuilder
+            ->setField('source_code')
+            ->setConditionType('eq')
+            ->setValue([$storeSourceCode])
+            ->create();
+
+        $defaultCodeFilter = $this->filterBuilder
+            ->setField('source_code')
+            ->setConditionType('eq')
+            ->setValue('default')
+            ->create();
+
+        $filterGroup2 = $this->filterGroupBuilder
+            ->setFilters([$sourceCodeFilter, $defaultCodeFilter])
+            ->create();
+
+        $this->searchCriteriaBuilder->setFilterGroups([$filterGroup1, $filterGroup2]);
+        $searchCriteria = $this->searchCriteriaBuilder->create();
 
+        $sourceItems = $this->sourceItemRepository->getList($searchCriteria)->getItems();
```

Comment:
Getting the in stock qty from the source item would probably be fine in most cases, but there are certain situations where it might not be accurate. 

As far as I know, the source items stores the "on hand" quantity for a SKU. However, this might be different from the quantity that's available for the customer to buy. Some things that might affect available quantity include open reservations or out-of-stock threshold settings. 

During the indexing process, some of that is taken into account (not reservations) and the available quantity is written to the index tables (inventory_stock_X). Ideally for this, we want to fetch quantity from the index tables for the stocks we're interested in. 

You can use [GetStockSourceLinksInterface](https://github.com/magento/inventory/blob/1.2-develop/InventoryApi/Api/GetStockSourceLinksInterface.php) to get the stock items for the provided source code. Then you can use [GetProductSalableQtyInterface](https://github.com/magento/inventory/blob/1.2-develop/InventorySalesApi/Api/GetProductSalableQtyInterface.php) to get the salable quantity for the SKU. If you take a look at the [implementation](https://github.com/magento/inventory/blob/423c7e5c5049c1e916cea4f595618e2a96ea765a/InventorySales/Model/GetProductSalableQty.php#L73) for that interface, you'll see it takes into account reservations.

There can be multiple stock items associated with one source, but I have yet to see that on an actual project I've worked on. But it might be good to sum up the available quantity for all stock items in this case.